UnitTests/UnitTests.FluentSharp.CoreLib/ExtensionMethods/IO/Test_IO_ExtensionMethods_Folders.cs

```diff
@@ -0,0 +1,306 @@
+            (null as string).folder_Create_File(filename, content).assert_Is_Null();
+            randomFilePath.folder_Create_File(filename, content).assert_Is_Null();
+            //cleanup
+            temporaryFolderPath.folder_Delete_Files().assert_Is_True();
+        }
+
+        [Test]
+        public void folder_Delete_Files()
+        {
+            var filename = "text.txt";
+            var content = "Lorem ipsum dolor";
+
+            (null as string).folder_Delete_Files().assert_Is_False();
+            "".folder_Delete_Files().assert_Is_False();
+            temporaryFolderPath.folder_Delete_Files().assert_Is_False();
+            temporaryFolderPath.folder_Create_File(filename, content).fileExists().assert_Is_True();
+            temporaryFolderPath.folder_Delete_Files().assert_Is_True();
+            randomFilePath.file_Not_Exists().assert_Is_True();
+            randomFilePath.folder_Delete_Files().assert_Is_False();
+        }
+
+        [Test]
+        public void folder_Exists()
+        {
+            (null as string).folder_Exists().assert_Is_False();
+            "".folder_Exists().assert_Is_False();
+            temporaryFolderPath.folder_Exists().assert_Is_True();
+            var newFolder = temporaryFolderPath.mapPath("newFolder").create_Folder();
+            newFolder.folder_Exists().assert_Is_True();
+            newFolder.delete_Folder().assert_Is_True();
+            rootDrive.folder_Exists().assert_Is_True();
+        }
+
+        [Test]
+        public void folder_Not_Exists()
+        {
+            (null as string).folder_Not_Exists().assert_Is_True();
+            "".folder_Not_Exists().assert_Is_True();
+            temporaryFolderPath.folder_Not_Exists().assert_Is_False();
+            randomFilePath.folder_Delete_Files().assert_Is_False();
+            temporaryFolderPath.pathCombine("something").folder_Not_Exists().assert_Is_True();
+            rootDrive.folder_Not_Exists().assert_Is_False();
+        }
+
+        [Test]
+        [UnitTestMethodReference("folder_Exists")]
+        public void folderExists()
+        {
+            MethodBase method = MethodBase.GetCurrentMethod();
+            UnitTestMethodReferenceAttribute attr = (UnitTestMethodReferenceAttribute)method.GetCustomAttributes(typeof(UnitTestMethodReferenceAttribute), true)[0];
+            attr.MethodToInvoke.invoke();
+        }
+
+        [Test]
+        public void dirExists()
+        {
+            (null as string).dirExists().assert_Is_False();
+            "".dirExists().assert_Is_False();
+            temporaryFolderPath.dirExists().assert_Is_True();
+            randomFilePath.folder_Delete_Files().assert_Is_False();
+            rootDrive.dirExists().assert_Is_True();
+            var newFolder = temporaryFolderPath.mapPath("newFolder").create_Folder();
+            newFolder.dirExists().assert_Is_True();
+            newFolder.delete_Folder().assert_Is_True();
+        }
```

Comment:
Isn't this one of the cases where the UnitTestMethodReferenceAttribute can be used?